lib/improver/tests/orographic_enhancement/test_OrographicEnhancement.py

```diff
@@ -0,0 +1,839 @@
+                                    [50., 80., 100., 200., 250.],
+                                    [50., 80., 100., 200., 250.]])
+        self.plugin.topography = iris.cube.Cube(
+            topography_data, long_name="topography", units="m",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+        humidity_data = np.full((5, 5), 0.9)
+        humidity_data[1, 3] = 0.5
+        self.plugin.humidity = iris.cube.Cube(
+            humidity_data, long_name="relhumidity", units="1",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+        self.plugin.vgradz = np.full((5, 5), 0.01)
+        self.plugin.vgradz[3:, :] = 0.
+
+    def test_basic(self):
+        """Test output is array"""
+        result = self.plugin._generate_mask()
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_values(self):
+        """Test output mask is correct"""
+        expected_output = np.array([[True, True, False, False, False],
+                                    [False, False, False, True, False],
+                                    [False, False, False, False, False],
+                                    [True, True, True, True, True],
+                                    [True, True, True, True, True]])
+        result = self.plugin._generate_mask()
+        self.assertArrayEqual(result, expected_output)
+
+
+class Test__site_orogenh(IrisTest):
+    """Test the _site_orogenh method"""
+
+    def setUp(self):
+        """Set up and populate a plugin instance"""
+        x_coord = DimCoord(np.arange(3), 'projection_x_coordinate',
+                           units='km')
+        y_coord = DimCoord(np.arange(3), 'projection_y_coordinate',
+                           units='km')
+
+        temperature = np.array([[277.1, 278.2, 277.7],
+                                [278.6, 278.4, 278.9],
+                                [278.9, 279.0, 279.6]])
+        humidity = np.array([[0.74, 0.85, 0.94],
+                             [0.81, 0.82, 0.91],
+                             [0.86, 0.93, 0.97]])
+        svp = np.array([[813.6, 878.0, 848.3],
+                        [903.2, 890.5, 922.2],
+                        [922.1, 928.4, 967.9]])
+
+        self.plugin = OrographicEnhancement()
+        self.plugin.temperature = iris.cube.Cube(
+            temperature, long_name="temperature", units="kelvin",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+        self.plugin.humidity = iris.cube.Cube(
+            humidity, long_name="relhumidity", units="1",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+        self.plugin.svp = iris.cube.Cube(
+            svp, long_name="saturation_vapour_pressure", units="Pa",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+        self.plugin.vgradz = np.array([[0.02, 0.08, 0.2],
+                                       [0.06, 0.12, 0.22],
+                                       [0.08, 0.16, 0.23]])
```

Comment:
Please can you set one of the vgradz values to be negative to simulate a down-slope flow which should be set to zero by the np.where statement at the end of _site_orogenh().

Reply:
Done